rust/private/transitions.bzl

```diff
@@ -64,3 +64,49 @@ with_import_macro_bootstrapping_mode = rule(
         ),
     },
 )
+
+def _without_process_wrapper_transition_impl(_settings, _attr):
+    """This transition allows rust_* rules to invoke rustc without process_wrapper."""
+    return {
+        "//command_line_option:compilation_mode": "opt",
```

Comment:
I'm sorry for that, this was a wrong suggestion - there is an 'exec' transition already happening when building the process wrapper, and that's where compilation mode is set. And since there is `--host_compilation_mode`, the process wrapper should respect that. I think I can edit the file so you don't have to.